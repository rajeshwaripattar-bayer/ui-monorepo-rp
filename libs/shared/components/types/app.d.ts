export interface Notification {
  open: boolean
  timeout?: number
  message: string
  dismissButtonLabel?: string
  actionButtonProps?: {
    label: string
    onAction: () => void
  }
}
