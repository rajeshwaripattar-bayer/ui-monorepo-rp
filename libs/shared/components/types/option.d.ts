export type Option = { value: string; text: string }
export type ApplyAllProps = { label?: string; applied?: boolean }
