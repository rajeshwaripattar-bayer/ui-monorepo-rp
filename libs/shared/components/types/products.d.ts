import { DiscDiscount, PriceValueAndCurrency } from '@gc/types'
import { Strategy } from './discounts'

export interface ProductWithPrice {
  canView?: boolean
  canOrder?: boolean
  cropCode: string
  cropName: string
  name: string
  code: string
  quantity: number
  netQuantity: number
  totalDiscount: number
  discounts?: { programName: string; strategy: Strategy }[]
  discountedUnitPrice: number
  unitPrice: number
  subTotal: number
  warehouse: { value: string; text: string }
  statusCounts?: {
    status: string
    count: number
    isBold?: boolean
  }[]
}

export interface PriceAndDiscountSummary {
  grossPrice: number
  discount: number
  discountPercentage: number
  discounts: {
    name: string
    value: number
  }[]
  netPrice: number
  discretionaryDiscounts: DiscDiscount[]
  totalDiscretionaryDiscount: PriceValueAndCurrency
  grossPriceDiscretionaryDiscount: PriceValueAndCurrency
  netPriceDiscretionaryDiscount: PriceValueAndCurrency
}

export interface ProductsByCrop {
  crop: string
  products: ProductWithPrice[]
  avgPrice: number
  summary: PriceAndDiscountSummary
}

export interface QuantityUpdateRequest {
  quantity: string | number
  productWithPrice: ProductWithPrice
  onlyCache?: boolean
  contingencyType?: 'alert' | 'dialog'
}

export interface ReturnedProducts {
  cropCode: string
  cropName: string
  name: string
  code: string
  quantity: number
  warehouseName?: string
  UOM: string
  masterOrderNumber?: string
}
export interface ReturnedProductsByCrop {
  crop: string
  products: ReturnedProducts[]
}

export type StockOrderProduct = {
  name: string
  warehouse: { value: string; text: string }
  quantity: number
  confirmedQuantity: number
  unconfirmedQuantity: number
}

export type StockOrderProductsByCrop = {
  crop: string
  count: number
  products: StockOrderProduct[]
}
