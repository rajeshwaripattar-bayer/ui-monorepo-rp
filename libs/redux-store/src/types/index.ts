import type { ExtractReducerState, EnhancedStore } from '@gc/types'
import { ccApi, middlewareApi, fgCcApi } from '../rtk-apis'
import type { AppState, CartState, FiltersState } from '../slices'

export type GlobalRootState = {
  app: AppState
  cart: CartState
  filters: FiltersState
  [ccApi.reducerPath]: ExtractReducerState<typeof ccApi.reducer>
  [fgCcApi.reducerPath]: ExtractReducerState<typeof fgCcApi.reducer>
  [middlewareApi.reducerPath]: ExtractReducerState<typeof middlewareApi.reducer>
}

export type GlobalAppStore = EnhancedStore<GlobalRootState>
export type GlobalAppDispatch = GlobalAppStore['dispatch']
