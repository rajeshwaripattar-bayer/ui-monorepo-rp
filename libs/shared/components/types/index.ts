export * from './app'
export * from './option'
export * from './products'
