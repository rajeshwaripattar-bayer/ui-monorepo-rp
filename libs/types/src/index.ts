export * from './accounts'
export * from './aem'
export * from './shared'
export * from './orderDetails'
export * from './stockOrder'
export * from './api'
export * from './app'
export * from './cart'
export * from './ccApi'
export * from './configData'
export * from './discounts'
export * from './domainDef'
export * from './farmers'
export * from './helpers'
export * from './locale'
export * from './orders'
export * from './products'
export * from './programs'
export * from './quoteDetails'
export * from './quotes'
export * from './redux'
export * from './deliveries'
export * from './selectedAccount'
export * from './user'
export * from './filter'
export * from './shipmentDeliveries'
export * from './fgCcApi'
export * from './addressInfo'

