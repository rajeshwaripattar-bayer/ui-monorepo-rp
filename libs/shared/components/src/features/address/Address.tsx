import { AddressInfo } from '@gc/types'
import styles from './Address.module.scss'
import { Typography } from '@element/react-typography'
import { useTranslation } from 'react-i18next'

/* eslint-disable-next-line */
export interface AddressProps {
  address: AddressInfo
  typographyType?: string
  className?: string
}

export function Address(props: AddressProps) {
  const { address, typographyType = 'body2', className } = props
  const { t } = useTranslation()
  return (
    <div className={className}>
      <Typography type={typographyType} className={styles.address}>
        <label>{address.addressee}</label>
        <label>{address.line1}</label>
        <label>{address.line2 && address.line2}</label>
        <label>{`${address.town}, ${address.region.isocodeShort} ${address.postalCode}`}</label>
        <label>{address.phone && `${t('common.telephone.label')}: ${address.phone}`}</label>
      </Typography>
    </div>
  )
}

export default Address
