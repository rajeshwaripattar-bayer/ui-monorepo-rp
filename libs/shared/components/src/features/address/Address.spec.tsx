import { render } from '@testing-library/react'
import { AddressInfo } from '@gc/types'
import Address from './Address'

describe('Address', () => {
  let addressInfo: AddressInfo
  beforeEach(() => {
    addressInfo = {
      addressee: 'CARL GREEN WAREHOUSE',
      line1: '2344 Test Address Dr',
      town: 'Kansas City',
      region: {
        countryIso: 'US',
        isocode: 'KS',
        isocodeShort: 'KS',
        name: 'Kansas'
      },
      postalCode: '66062'
    }
  })

  it('should render successfully', () => {
    const { baseElement } = render(<Address address={addressInfo} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render addressline 2 successfully, if passed', () => {
    addressInfo.line2 = 'Round Cr'
    const { baseElement } = render(<Address address={addressInfo} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render phone successfully, if passed', () => {
    addressInfo.phone = '574-556-2345'
    const { baseElement } = render(<Address address={addressInfo} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render with expected text style based on typography being passed', () => {
    const { container } = render(<Address address={addressInfo} typographyType='display4' />)
    expect(container.getElementsByClassName('mdc-typography lmnt mdc-typography--headline4').length).toBe(1)
  })

  it('should render with body2 text typograhy style by default', () => {
    const { container } = render(<Address address={addressInfo} />)
    expect(container.getElementsByClassName('mdc-typography lmnt mdc-typography--body2').length).toBe(1)
  })

  it('should render in expected style class if className is passed', () => {
    const { container } = render(<Address address={addressInfo} className='address_style' />)
    expect(container.getElementsByClassName('address_style').length).toBe(1)
  })
})
