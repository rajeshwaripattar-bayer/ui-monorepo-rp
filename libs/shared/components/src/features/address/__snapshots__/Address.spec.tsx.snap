// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address should render addressline 2 successfully, if passed 1`] = `
<body>
  <div>
    <div>
      <span
        class="mdc-typography lmnt mdc-typography--body2 address"
      >
        <label>
          CARL GREEN WAREHOUSE
        </label>
        <label>
          2344 Test Address Dr
        </label>
        <label>
          Round Cr
        </label>
        <label>
          Kansas City, KS 66062
        </label>
        <label />
      </span>
    </div>
  </div>
</body>
`;

exports[`Address should render phone successfully, if passed 1`] = `
<body>
  <div>
    <div>
      <span
        class="mdc-typography lmnt mdc-typography--body2 address"
      >
        <label>
          CARL GREEN WAREHOUSE
        </label>
        <label>
          2344 Test Address Dr
        </label>
        <label />
        <label>
          Kansas City, KS 66062
        </label>
        <label>
          common.telephone.label: 574-556-2345
        </label>
      </span>
    </div>
  </div>
</body>
`;

exports[`Address should render successfully 1`] = `
<body>
  <div>
    <div>
      <span
        class="mdc-typography lmnt mdc-typography--body2 address"
      >
        <label>
          CARL GREEN WAREHOUSE
        </label>
        <label>
          2344 Test Address Dr
        </label>
        <label />
        <label>
          Kansas City, KS 66062
        </label>
        <label />
      </span>
    </div>
  </div>
</body>
`;
